cpp/tests/spatial/point_in_polygon_test.cpp

```diff
@@ -58,3 +58,137 @@ TYPED_TEST(PointInPolygonTest, Empty)
 
   expect_columns_equal(expected, actual->view(), verbosity);
 }
+
+template <typename T>
+struct PointInPolygonUnsupportedTypesTest : public BaseFixture {
+};
+
+using UnsupportedTestTypes = RemoveIf<ContainedIn<TestTypes>, NumericTypes>;
+TYPED_TEST_CASE(PointInPolygonUnsupportedTypesTest, UnsupportedTestTypes);
+
+TYPED_TEST(PointInPolygonUnsupportedTypesTest, UnsupportedPointType)
+{
+  using T = TypeParam;
+
+  auto test_point_xs     = wrapper<T>({0.0, 0.0});
+  auto test_point_ys     = wrapper<T>({0.0});
+  auto poly_offsets      = wrapper<cudf::size_type>({0});
+  auto poly_ring_offsets = wrapper<cudf::size_type>({0});
+  auto poly_point_xs     = wrapper<T>({0.0, 1.0, 0.0, -1.0});
+  auto poly_point_ys     = wrapper<T>({1.0, 0.0, -1.0, 0.0});
+
+  EXPECT_THROW(
+    cuspatial::point_in_polygon(
+      test_point_xs, test_point_ys, poly_offsets, poly_ring_offsets, poly_point_xs, poly_point_ys),
+    cuspatial::logic_error);
+}
+
+template <typename T>
+struct PointInPolygonUnsupportedChronoTypesTest : public BaseFixture {
+};
+
+TYPED_TEST_CASE(PointInPolygonUnsupportedChronoTypesTest, ChronoTypes);
```

Comment:
I can't remember why we have unsupported types tests specifically for Chrono types but not for other non-numerical types (e.g. nested types). Is it necessary to test these at all? Every type we test increases compile time, so I wonder if we can just test one unsupported type as a proxy for all the others? 

Don't necessarily need to decide this in the present PR!